vault/models/vault_entry.py

```diff
@@ -73,6 +74,36 @@ def _compute_complete_name(self):
             else:
                 rec.complete_name = rec.name
 
+    @api.model
+    def search_panel_select_range(self, field_name, **kwargs):
+        """We add the following contexts related to searchpanel:
+        - entry_short_name: Show just the name instead of full path.
+        - from_search_panel: It will be used to overwrite domain.
+        """
+        return super(
+            VaultEntry,
+            self.with_context(from_search_panel=True, entry_short_name=True),
+        ).search_panel_select_range(field_name, **kwargs)
```

Comment:
In the kwargs is a limit parameter with v14+. You might want to set it to None here otherwise the default 200 comes from the JS.

Reply:
![too-many](https://user-images.githubusercontent.com/39951254/187925339-9535bad1-3cf8-4e5c-97aa-87f7bb86ca6d.png)

Doesn't fit my tests. You are right limit is None in python but it's getting called from JS and the DEFAULT_LIMIT is set there https://github.com/odoo/odoo/blob/14.0/addons/web/static/src/js/views/search_panel_model_extension.js#L13

Reply:
Oops, I didn't see that odoo had moved the limit to js. 
Thanks, it's changed.